homeassistant/components/pegel_online/__init__.py

```diff
@@ -0,0 +1,71 @@
+"""The PEGELONLINE component."""
+from __future__ import annotations
+
+import logging
+
+from aiopegelonline import CONNECT_ERRORS, PegelOnline
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import (
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import (
+    CONF_STATION,
+    DOMAIN,
+    MIN_TIME_BETWEEN_UPDATES,
+)
+from .model import PegelOnlineData
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up PEGELONLINE entry."""
+    station_uuid = entry.data[CONF_STATION]
+    name = entry.title
+
+    _LOGGER.debug("Setting up station with uuid %s", station_uuid)
+
+    api = PegelOnline(async_get_clientsession(hass))
+    station = await api.async_get_station_details(station_uuid)
+
+    async def async_update_data() -> PegelOnlineData:
+        """Fetch data from API endpoint."""
+        try:
+            current_measurement = await api.async_get_station_measurement(station_uuid)
+        except CONNECT_ERRORS as err:
+            raise UpdateFailed(f"Failed to communicate with API: {err}") from err
+
+        return {"current_measurement": current_measurement}
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=name,
+        update_method=async_update_data,
+        update_interval=MIN_TIME_BETWEEN_UPDATES,
+    )
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = (coordinator, station)
+
+    await coordinator.async_config_entry_first_refresh()
+
```

Comment:
We should not set hass data before the first refresh 

```suggestion
    await coordinator.async_config_entry_first_refresh()

    hass.data.setdefault(DOMAIN, {})
    hass.data[DOMAIN][entry.entry_id] = (coordinator, station)
```